distributed/core.py

```diff
@@ -109,6 +109,131 @@ def _expects_comm(func: Callable) -> bool:
     return False
 
 
+class AsyncTaskGroup:
+    """Collection tracking all currently running asynchronous tasks within a group"""
+
+    #: If True, the group is closed and does not allow adding new tasks.
+    closed: bool
+
+    def __init__(self) -> None:
+        self.closed = False
+        self._lock = threading.Lock()
+        self._ongoing_tasks: set[asyncio.Task] = set()
+
+    def schedule(self, coro: Coroutine) -> asyncio.Task | None:
+        """Schedules a coroutine object to be executed as an `asyncio.Task`.
+
+        Parameters
+        ----------
+        coro : Coroutine
+            Coroutine object to schedule.
+
+        Returns
+        -------
+        asyncio.Task | None
+            The scheduled Task object, or None if the group is closed.
+        """
+        with self._lock:
```

Comment:
this shouldn't be needed - as `asyncio.create_task(coro)` can only work from the thread that created AsyncTaskGroup

Reply:
Done